gtubt/lib/ui/pages/calendar_page.dart

```diff
@@ -1,13 +1,105 @@
+import 'package:GTUBT/ui/blocs/calendar_bloc/bloc.dart';
+import 'package:GTUBT/ui/style/text_styles.dart';
+import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
+import 'package:GTUBT/models/event.dart';
+import 'package:GTUBT/ui/style/color_sets.dart';
+
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class CalendarPage extends StatefulWidget {
+  @override
+  _CalendarPageState createState() => _CalendarPageState();
+}
+
+class _CalendarPageState extends State<CalendarPage> {
+  final List<String> months = [
+    'January',
+    'February',
+    'March',
+    'April',
+    'May',
+    'June',
+    'July',
+    'August',
+    'September',
+    'October',
+    'November',
+    'December'
+  ];
+
+  @override
+  void initState() {
+    super.initState();
+    BlocProvider.of<CalendarPageBloc>(context).add(FetchEvents());
+  }
 
-class CalendarPage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    return Container(
-      padding: EdgeInsets.all(30),
-      child: Text(
-        "Calendar Page",
-      ),
+    return BlocConsumer<CalendarPageBloc, CalendarPageState>(
+      listener: (context, state) {},
+      builder: (context, CalendarPageState state) {
+        return Scaffold(
+            backgroundColor: ColorSets.pageBackgroundColor,
+            body: _buildBody(state));
+      },
     );
   }
+
+  Widget _buildBody(CalendarPageState state) {
+    Widget body;
+    if (state is EventsLoadingError) {
+      body = Text('An Error Occur' // state.failure.toString()
```

Comment:
bu burada olmamali, bloc consumer'in listener'inda olmali ve NotificationFactory kullanilarak hata gosterilmeli. ayrica `buildWhen` kullanarak error state ise sayfa yeniden render edilmesi engellenmeli